src/App.tsx

```diff
@@ -1,223 +1,272 @@
+  const [tempTodo, setTempTodo] = useState<Todo>({
+    id: 0,
+    title: '',
+    userId: 0,
+    completed: false,
+  });
+
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const changeErrorStatus = () => setErrorStatus(false);
+
+  async function loadTodos() {
+    if (user) {
+      const todosFromServer = await getTodos(user.id);
+
+      setTodos(todosFromServer);
+    }
+  }
+
+  const addTodoOnServer = async (todoTitle: string) => {
+    if (user) {
+      try {
+        setTempTodo({
+          id: 0,
+          title: todoTitle,
+          userId: user.id,
+          completed: false,
+        });
+        setIsAdding(true);
+
+        const newTodo = await addTodo({
+          title: todoTitle,
+          userId: user.id,
+          completed: false,
+        });
+
+        setTodos(currTodos => [...currTodos, newTodo]);
+        setIsAdding(false);
+      } catch {
+        setErrorMessage('Unable to add a todo');
+        setErrorStatus(true);
+        setIsAdding(false);
+        setTempTodo({
+          id: 0,
+          title: '',
+          userId: 0,
+          completed: false,
+        });
+      }
+    }
+  };
+
+  const removeTodoFromServer = async (todoId: number) => {
+    try {
+      setIsRemoving(prevIds => [...prevIds, todoId]);
+      await removeTodo(todoId);
+
+      setTodos(currTodos => currTodos.filter(todo => todo.id !== todoId));
+      setIsRemoving(prevIds => prevIds.filter(id => id !== todoId));
+    } catch {
+      setErrorMessage('Unable to delete a todo');
+      setErrorStatus(true);
+    }
+  };
+
+  const removeCompletedTodos = async () => {
+    try {
+      await Promise.all(todos.map(todo => {
+        if (todo.completed) {
+          setIsRemoving(prevIds => [...prevIds, todo.id]);
+          removeTodo(todo.id);
+        }
+
+        return todo;
+      }));
+
+      setTimeout(() => {
+        setTodos(currTodos => currTodos.filter(todo => !todo.completed));
+      }, 500);
+    } catch {
+      setErrorMessage('Unable to delete todos');
+      setErrorStatus(true);
+    }
+  };
+
+  const changeTodoOnServer = async (todoId: number, completed: boolean) => {
+    try {
+      setIsRemoving(prevIds => [...prevIds, todoId]);
+      await changeTodo({ completed }, todoId);
+
+      setTodos(currTodos => currTodos.map((todo: Todo) => {
+        if (todo.id === todoId) {
+          const changedTodo = todo;
+
+          changedTodo.completed = completed;
+
+          return changedTodo;
+        }
+
+        return todo;
+      }));
+      setIsRemoving([]);
+    } catch {
+      setErrorMessage('Unable to update a todo');
+      setErrorStatus(true);
+    }
+  };
+
+  const changeTitleOnServer = async (todoId: number, title: string) => {
+    try {
+      setIsRemoving(prevIds => [...prevIds, todoId]);
+      await changeTodo({ title }, todoId);
+
+      setTodos(currTodos => currTodos.map((todo: Todo) => {
```

Comment:
here will be also good to get actual data from server

Reply:
done